### PR TITLE
File locking

### LINKED ISSUE #10012: Implement file locking
Ref: https://github.com/nextcloud/files-clients/issues/2

### PR DESCRIPTION
**Pull request by AlvaroBrey**
Closes #10012
Needs https://github.com/nextcloud/android-library/pull/851

- [x] Show lock status
    - [x] Lock status on file list
    - ~~[] Lock status on details view~~ will show up as activity
    - [x] Lock details popup (owner, time, type, TTL) 
- [x] Allow manual lock/unlock
    - [x] Only allow unlock if lock owner is user and lock is manual type 
- [x] Prevent write operations on locked files
    - [x]  Prevent delete
    - [x] Prevent move (allow copy)
- [x] Tests written, or not not needed

### CODE REVIEW

File: app/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 1]
```diff
@@ -917,6 +918,12 @@ public void toggleEncryption(OCFile file, boolean shouldBeEncrypted) {
         }
     }
 
+    public void toggleFileLock(OCFile file, boolean shouldBeLocked) {
+        if (file.isLocked() != shouldBeLocked) {
+            EventBus.getDefault().post(new FileLockEvent(file.getRemotePath(), shouldBeLocked));
```
Comment (AlvaroBrey) [inline app/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java:923]: Note for reviewers: I am unsure about this approach. I thought about using OperationsService, but it does not support NextcloudClient I think.

File: app/src/main/java/com/owncloud/android/ui/events/FileLockEvent.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Nextcloud Android client application
+ *
+ * @author Álvaro
+ * Copyright (C) 2017 Mario Danic
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/owncloud/android/ui/events/FileLockEvent.kt:5]: 2022 Alvaro?

File: app/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 1 above)
Comment (tobiasKaminsky) [reply app/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java:923]: Fine for now, as we use EventBus also in lots of other places.
Once we feel uncomfortable with it, we can find a better solution and change it all in one rush.

File: app/src/main/res/drawable/ic_locked_dots_small.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tobiasKaminsky) [inline app/src/main/res/drawable/ic_locked_dots_small.xml:1]: License?

**Review by tobiasKaminsky — APPROVED** [review]

**Pull request merged**